### PR TITLE
CNV13829: Adding Comprehensive 4.10 Release Notes

### PR DESCRIPTION
**Pull request by mgarrell777**
This PR is for the 4.10 Release Notes [CNV-13829](https://issues.redhat.com/browse/CNV-13829)

CP: 4.10

Preview: https://deploy-preview-42530--osdocs.netlify.app/openshift-enterprise/latest/virt/virt-4-10-release-notes

Previous PR: https://github.com/openshift/openshift-docs/pull/41320

### CODE REVIEW

**Review by dominikholler — CHANGES_REQUESTED** [review]

**Comment by lkladnit** [comment]
A new user left a comment. This user must be approved by a Netlify team owner before comments can be displayed.

[Approve this user](https://app.netlify.com/teams/vikram-redhat/reviewers)

File: virt/virt-4-10-release-notes.adoc [hunk 1]
```diff
@@ -234,7 +242,53 @@ metadata:
...
+
+* If you clone more than 100 VMs using the `csi-clone` cloning strategy, then the Ceph CSI might not purge the clones. Manually deleting the clones might not work either.
+** As a workaround, you can restart the `ceph-mgr` to purge the VM clones. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060371[*BZ#2060371*])
 
-* If a single node contains more than 50 images, pod scheduling might be imbalanced across nodes. This is because the list of images on a node is shortened to 50 by default.
-** As a workaround, you can disable the image limit by xref:../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-managing[editing the `KubeletConfig` object] and setting the value of `nodeStatusMaxImages` to `-1`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1984442[*BZ#1984442*])
+* If you clone more than 100 VMs on Red Hat Ceph Storage or Red Hat OpenShift Data Foundation Storage at the same time, then the cloning process might stall and fail.
+** As a workaround, create VMs in smaller bulks. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060374[*BZ#2060374*])
```
[the remark this answers is not in the document]
Comment (jeniferh) [reply virt/virt-4-10-release-notes.adoc:294]: Unfortunately due to other cleanup related bugs, even "create VMs in smaller bulks" is not a proper workaround. This workaround should be changed, suggestion:

As a workaround, when using Red Hat Ceph Storage or Red Hat OpenShift Data Foundation Storage, set the default cloning strategy to "copy" which will perform a host-assisted copy. To do so, edit the storage profile as described here and set `cloneStrategy: copy` (link:https://docs.openshift.com/container-platform/4.10/virt/virtual_machines/virtual_disks/virt-creating-data-volumes.html#virt-customizing-storage-profile-default-cloning-strategy_virt-creating-data-volumes)

-- I believe that ocp doc link should work once the CNV sections go live?

File: virt/virt-4-10-release-notes.adoc [hunk 2]
```diff
@@ -234,7 +242,53 @@ metadata:
...
+** As a workaround, non-privileged users can watch the `NAD Network Settings` screen to confirm the VM creation process is successful. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2056420[*BZ#2056420*])
+
+* A non-privileged user cannot add disks to a VM due to RBAC rules.
+** As a workaround, manually add the RBAC rule to allow specific users to add disks. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2056421[*BZ#2056421*])
+
+* The web console does not display virtual machine templates that are deployed to a custom namespace. Only templates deployed to the default namespace will display in the web console.
+** As a workaround, avoid deploying templates to a custom namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2054650[*BZ#2056623*])
+
+* If you clone more than 100 VMs using the `csi-clone` cloning strategy, then the Ceph CSI might not purge the clones. Manually deleting the clones might not work either.
+** As a workaround, you can restart the `ceph-mgr` to purge the VM clones. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060371[*BZ#2060371*])
```
Comment (dominikholler) [inline virt/virt-4-10-release-notes.adoc:291]: Bugzilla 2055595 should be referenced here

**Review by dominikholler — CHANGES_REQUESTED** [review]

File: virt/virt-4-10-release-notes.adoc (hunk 2 above)
Comment (mgarrellRH) [reply virt/virt-4-10-release-notes.adoc:291]: Can you please add Doc Text to Bug 2055595 so I can include it?

**Comment by sagidlow** [comment]
Test comment

File: virt/virt-4-10-release-notes.adoc (hunk 2 above)
Comment (dominikholler) [reply virt/virt-4-10-release-notes.adoc:291]: > @dominikholler Can you please add Doc Text to Bug 2055595 so I can include it?

@mgarrellRH This would be not correct. RHBZ 2060371 is the know issue about what RHBZ 2055595 means for OpenShift Virtualization, while 2055595 is about a bug in ceph, which is independent from OpenShift Virtualization.

File: virt/virt-4-10-release-notes.adoc [hunk 3]
```diff
@@ -146,33 +156,31 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
-* Running virtual machines that cannot be live migrated might block an {product-title} cluster upgrade. This includes virtual machines that use hostpath provisioner storage or SR-IOV network interfaces.
+* You can now successfully import a VM image when using a cluster-wide proxy configuration. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2046271[*BZ#2046271*])
+
+[id="virt-4-10-known-issues"]
+== Known issues
+
+* Editing a virtual machine fails if the VM references a deleted template that was provided by {VirtProductName} before version 4.8. In {VirtProductName} 4.8 and later, deleted {VirtProductName}-provided templates are automatically recreated by the {VirtProductName} Operator.
+
+* If your {product-title} cluster uses OVN-Kubernetes as the default Container Network Interface (CNI) provider, you cannot attach a Linux bridge or bonding to the default interface of a host because of a change in the host network topology of OVN-Kubernetes.
+** As a workaround, you can use a secondary network interface connected to your host, or switch to the OpenShift SDN default CNI provider. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1885605[*BZ#1885605*])
```
Comment (ctomasko) [inline virt/virt-4-10-release-notes.adoc:181]: 1885605 is resolved. Remove from known issues

File: virt/virt-4-10-release-notes.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (dankenigsberg) [reply virt/virt-4-10-release-notes.adoc:294]: @mgarrellRH in more generic terms: release notes should never point to a doc bug, only to the underlying code bug that is either open or has been fixed.

[the remark this answers is not in the document]
Comment (jeniferh) [reply virt/virt-4-10-release-notes.adoc]: looks good, thank you!

**Review by dominikholler — APPROVED** [review]
lgtm

File: virt/virt-4-10-release-notes.adoc [hunk 4]
```diff
@@ -145,39 +156,37 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
 
-* If your {product-title} cluster uses OVN-Kubernetes as the default Container Network Interface (CNI) provider, you cannot attach a Linux bridge or bonding to the default interface of a host because of a change in the host network topology of OVN-Kubernetes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1885605[*BZ#1885605*])
+* If you provision a virtual machine disk, {VirtProductName} now allocates a PVC that is just large enough to accommodate the requested disk size, rather than a `KubePersistentVolumeFillingUp` alert occurring for each VM disk PVC. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2039489[*BZ#2039489*])
 
-** As a workaround, you can use a secondary network interface connected to your host, or switch to the OpenShift SDN default CNI provider.
+* If you create a persistent hot-plug volume and a VMSnapshot, now you also create a VolumeSnapshot instance for each volume in the VMSnapshot. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042908[*BZ#2042908*])
```
Comment (sjhala-ccs) [inline virt/virt-4-10-release-notes.adoc:171]: Suggest rephrasing this as:

"You can now create a virtual machine snapshot for VMs with hot-plugged disks."

**Review by sjhala-ccs — COMMENTED** [review]
Reviewed everything except the Known Issues section. @ousleyp Can you please take a look at the known issues? Thanks.

**Review by ousleyp — COMMENTED** [review]
Reviewed the Known Issues section only. Looking good so far, just a few clarity/informational issues and some minor fixes.

File: virt/virt-4-10-release-notes.adoc [hunk 5]
```diff
@@ -196,44 +194,74 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
+  name: <provisioner_class>
+#   ...
+spec:
+  claimPropertySets:
+  - accessModes:
+    - ReadWriteOnce
+    volumeMode:
+      Filesystem
+  cloneStrategy:
+  copy <1>
```
[the remark this answers is not in the document]
Comment (jeniferh) [reply virt/virt-4-10-release-notes.adoc:214]: Functionally I don't think it matters, all on one line does make it easier to read

File: virt/virt-4-10-release-notes.adoc [hunk 6]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
+* YAML examples in the VM wizard are not always up to date with the latest information because the examples are hardcoded upstream. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055492[*BZ#2055492*])
+
+* If you clone more than 100 VMs using the `csi-clone` cloning strategy, then the Ceph CSI might not purge the clones. Manually deleting the clones can also fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055595[*BZ#2055595*])
+** As a workaround, you can restart the `ceph-mgr` to purge the VM clones.
+
+* A non-privileged user cannot use the Add Network Interface button on the `VM Network Interfaces` tab. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2056420[*BZ#2056420*])
+** As a workaround, non-privileged users can add additional network interfaces while creating the VM by using the VM wizard.
+
+* A non-privileged user cannot add disks to a VM due to RBAC rules.
+** As a workaround, manually add the RBAC rule to allow specific users to add disks. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2056421[*BZ#2056421*])
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:259]: Move this link to the end of the first bullet

File: virt/virt-4-10-release-notes.adoc [hunk 7]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
-+
-** In the web console, if you navigate to the *Virtualization* page and select *Create* -> *With YAML* the following error message is displayed:
-+
-[source,text]
-----
-The server doesn't have a resource type "kind: VirtualMachine, apiVersion: kubevirt.io/v1"
-----
-+
-*** As a workaround, edit the `VirtualMachine` manifest so the `apiVersion` is `kubevirt.io/v1alpha3`. For example:
+* In some instances, multiple virtual machines can mount the same PVC in read-write mode, which might result in data corruption. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1992753[*BZ#1992753*]))
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:219]: Remove the extra `)` at the end of the link

File: virt/virt-4-10-release-notes.adoc [hunk 8]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
+** As a workaround, avoid using a single PVC in read-write mode with multiple VMs
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:220]: Add a period after "multiple VMs"

File: virt/virt-4-10-release-notes.adoc [hunk 9]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
+
+* The Pod Disruption Budget (PDB) prevents pod disruptions for migratable virtual machine images. If the PDB detects pod disruption, then `openshift-monitoring` sends a `PodDisruptionBudgetAtLimit` alert every 60 minutes for virtual machine images that use the `LiveMigrate` eviction strategy. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2026733[*BZ#2026733*])
+** As a workaround, see xref:../monitoring/managing-alerts.adoc#silencing-alerts_managing-alerts[Silencing alerts].
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:223]: Please change this to `** As a workaround, xref:../monitoring/managing-alerts.adoc#silencing-alerts_managing-alerts[silence the alert].`

File: virt/virt-4-10-release-notes.adoc [hunk 10]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
-*** As a workaround, create the virtual machine from the CLI or upgrade to {VirtProductName} 4.8.
-+
-(link:https://bugzilla.redhat.com/show_bug.cgi?id=1977037[*BZ#1977037*])
 
-//Known issues discovered after 4.9 release
+* {VirtProductName} links a service account token in use by a pod to that specific pod. {VirtProductName} implements a service account volume by creating a disk image that contains a token. If you migrate a VM, then the service account volume becomes invalid. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2037611[*BZ#2037611*])
+** As a workaround, use user accounts rather than service accounts because user account tokens are not bound to a specific pod.
+
+* {VirtProductName} does not process multiple attempts to stop a VM. If a VM crashes during shutdown, then you cannot issue a new stop attempt using `--force --grace-period 0` flags and you cannot easily remove the VM from the UI. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2040766[*BZ#2040766*])
+** As a workaround, delete the virt-launcher pod directly.
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:243]: We usually put virt-launcher in backticks

File: virt/virt-4-10-release-notes.adoc [hunk 11]
```diff
@@ -196,44 +194,69 @@ $ oc annotate --overwrite -n openshift-cnv hyperconverged kubevirt-hyperconverge
...
+** As a workaround:
+. Remove `mediatedDevicesConfiguration` and `permittedHostDevices` from the `HyperConverged` CR.
+. Update both `mediatedDevicesConfiguration` and `permittedHostDevices` stanzas with the configuration you want to use.
 
-* If a single node contains more than 50 images, pod scheduling might be imbalanced across nodes. This is because the list of images on a node is shortened to 50 by default.
-** As a workaround, you can disable the image limit by xref:../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-managing[editing the `KubeletConfig` object] and setting the value of `nodeStatusMaxImages` to `-1`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1984442[*BZ#1984442*])
+* YAML examples in the VM wizard are not always up to date with the latest information because the examples are hardcoded upstream. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055492[*BZ#2055492*])
```
Comment (ousleyp) [inline virt/virt-4-10-release-notes.adoc:250]: The examples aren't hardcoded upstream; they might change upstream, and the hardcoded examples in the VM wizard won't change with them. I'd rephrase the sentence to:
```
* YAML examples in the VM wizard are hardcoded and do not always contain the latest upstream changes.
```

**Review by ousleyp — COMMENTED** [review]
I did a quick scan and noticed a few minor issues. Nothing else that I noticed is a dealbreaker so I'm happy to merge after these last few fixes. Thanks, Matthew!
